contrib/scalar/scalar.c

```diff
@@ -695,6 +695,45 @@ static char *remote_default_branch(const char *url)
 	return NULL;
 }
 
+static void strbuf_parentdir(struct strbuf *buf)
+{
+	for (; buf->len > 0 && !is_dir_sep(buf->buf[buf->len]); buf->len--)
+		; /* keep looking for parent directory */
+	buf->buf[buf->len] = '\0';
+}
+
+static int delete_enlistment(void)
+{
+	struct strbuf enlistment = STRBUF_INIT;
+
+	/* Compute the enlistment path (parent of the worktree) */
+	strbuf_addstr(&enlistment, the_repository->worktree);
+	strbuf_parentdir(&enlistment);
+
+#ifdef WIN32
+	/* Change current directory to one outside of the enlistment
+	   so that we may delete everything underneath it. */
+	if (is_inside_dir(enlistment.buf)) {
+		struct strbuf parent = STRBUF_INIT;
+		strbuf_addbuf(&parent, &enlistment);
+		strbuf_parentdir(&parent);
```

Comment:
An even more canonical way would probably be to employ `find_last_sep()` to find the position of the last slash, assigning `enlistment.buf[slash] = '\0';`, then `chdir(enlistment.buf)` and then assigning `enlistment.buf[slash] = '\\';` again.

But then, can we even delete the directory if we're inside it? Wouldn't the caller be blocking the removal of that directory?

Reply:
> But then, can we even delete the directory if we're inside it? Wouldn't the caller be blocking the removal of that directory?

But we'd be changing the directory to the parent of the enlistment dir, so we're not inside it?